web-api-shared/src/elastic.rs

```diff
@@ -367,10 +527,24 @@ pub enum Error {
         url: Url,
         error: String,
     },
-    /// Failed to serialize a requests or deserialize a response.
+    /// Failed to serialize a requests or deserialize a response: {0}
     Serialization(serde_json::Error),
     /// Given endpoint was not found: {0}
-    EndpointNotFound(&'static str),
+    ResourceNotFound(String),
+}
+
+pub trait NotFoundAsOptionExt<T> {
+    fn not_found_as_option(self) -> Result<Option<T>, Error>;
+}
+
+impl<T> NotFoundAsOptionExt<T> for Result<T, Error> {
+    fn not_found_as_option(self) -> Result<Option<T>, Error> {
+        match self {
+            Err(Error::ResourceNotFound(_)) => Ok(None),
+            Ok(value) => Ok(Some(value)),
+            Err(error) => Err(error),
```

Comment:
❓ shouldn't the happy path come first in a match?
```suggestion
        match self {
            Ok(value) => Ok(Some(value)),
            Err(Error::ResourceNotFound(_)) => Ok(None),
            Err(error) => Err(error),
```